app/models/two_factors_memorized_device.rb

```diff
@@ -0,0 +1,20 @@
+class TwoFactorsMemorizedDevice < ApplicationRecord
+  belongs_to :user
+
+  scope :active, -> do
+    where("created_at >= ?", self.active_duration.ago)
+  end
+
+  # expires when created at date is earlier than the date 30 days ago
+  scope :expired, -> do
+    where("created_at < ?", self.active_duration.ago)
+  end
+
+  def self.active_duration
+    30.days
```

Comment:
Move this to `GrdaWarehouse::Config` with a picker on `admin/configs` to choose, 0, 7, 15, 30 days.  Zero days would disable the ability to remember a device.